src/datatable.py

```diff
@@ -0,0 +1,192 @@
+
+                uwu = ""
+                if col == "fullgnomen":
+                    uwu = '<a href="/variant/{0}">{0}</a>'.format(val)
+                elif col == "dn_no":
+                    uwu = '<a href="/patient/{0}">{0}</a>'.format(val)
+                elif col == "gene":
+                    uwu = '<a href="/gene/{0}">{0}</a>'.format(val)
+                else:
+                    uwu = val
+                data_row.append(uwu)
+            data_rows.append(data_row)
+        output['data'] = data_rows
+        return output
+
+    def output_result_on_queried_fields(self):
+        output = {}
+        output['draw'] = str(int(self.request['draw']))
+        output['recordsTotal'] = str(self.records_total)
+        output['recordsFiltered'] = str(self.records_filtered)
+
+        data_rows = []
+        i = self.paging().start
+        for row in self.result_data:
+            data_row = []
+            i += 1
+            data_row.append(i)
+            for col, val in row.items():
+                uwu = ""
+                if col == "_id":
+                    if self.group_by == "fullgnomen":
+                        uwu = '<a href="/variant/{0}">{0}</a>'.format(val)
+                    elif self.group_by == "dn_no":
+                        uwu = '<a href="/patient/{0}">{0}</a>'.format(val)
+                    elif self.group_by == "gene":
+                        uwu = '<a href="/gene/{0}">{0}</a>'.format(val)
+                elif col == "total":
+                    uwu = val
+                else:
+                    # filtering like this doesn't work yet
+                    # uwu = '<a href="/all?{1}={0}">{0}</a>'.format(val, col)
+                    uwu = val
+                data_row.append(uwu)
+            data_rows.append(data_row)
+
+        output['data'] = data_rows
+        return output
+
+    def run_queries(self):
+        pages = self.paging()  # pages has 'start' and 'length' attributes
+        filter = self.filtering()  # the term entered into the datatable search
+        sorting = self.sorting()  # the document field chosen to sort
+
+        # get result from db
+        match = {"$match": filter}
+        group = {}
+        if self.group_by is not None:
+            group = {
+                "$group": {
+                    "_id": "$" + self.group_by,
+                    "total": {
+                        "$sum": 1
+                    }
+                }
+            }
+
+        if self.group_by == "fullgnomen":
+            group["$group"]["protein"] = {"$addToSet": "$pnomen"}
+        pipeline = []
+
+        if group:
+            pipeline.append(group)
+        if filter:
+            pipeline.append(match)
+        pipeline = pipeline + [{"$sort": sorting}, {"$skip": pages.start}]
+        if pages.length >= 0:
+            pipeline.append({"$limit": pages.length})
+        self.result_data = list(self.dbh[self.collection].aggregate(pipeline))
```

Comment:
With my (very limited) knowledge regarding classes and OOP,  I believe it is common and good practice to use getters and setters. I am a bit worried about changing class specific variables in different functions. What are your thoughts about this?

Reply:
see earlier comment:
I have made another issue #19, so that the next version needs to have a look.

Reply:
I think al the "getting" and "setting" is done using the datatables plugin (javascripts). [link](https://datatables.net/)
I think that the datainput is regulated by the datatables.py by sending queries to the mongoDB database. this is then returned to the datatables plugin to use as variables ect. This way it is always dynamic and every datatable can be made on the fly, and not all information is stored in a python dictionary.